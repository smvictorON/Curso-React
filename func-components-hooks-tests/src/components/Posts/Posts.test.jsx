import { render, screen } from '@testing-library/react';
import { Posts } from '.';

const mock = {
  posts: [
    {
      title: 'Titulo1',
      body: 'Corpo1',
      id: 1,
      cover: 'img/img1.png',
    },
    {
      title: 'Titulo2',
      body: 'Corpo2',
      id: 2,
      cover: 'img/img2.png',
    },
    {
      title: 'Titulo3',
      body: 'Corpo3',
      id: 3,
      cover: 'img/img3.png',
    },
  ],
};

describe('<Posts/>', () => {
  it('should render posts', () => {
    render(<Posts {...mock} />);

    expect(screen.getAllByRole('heading', { name: /Titulo/i })).toHaveLength(3);
    expect(screen.getAllByRole('img', { name: /Titulo/i })).toHaveLength(3);
    expect(screen.getAllByText(/Corpo/i)).toHaveLength(3);
    expect(screen.getByRole('img', { name: /Titulo2/i })).toHaveAttribute('src', 'img/img2.png');
  });

  it('should render with no posts', () => {
    render(<Posts />);

    expect(screen.queryByRole('heading', { name: /Titulo/i })).not.toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(<Posts {...mock} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
