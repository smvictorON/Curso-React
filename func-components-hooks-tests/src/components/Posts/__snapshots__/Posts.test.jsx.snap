// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts/> should match snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="Titulo1"
      src="img/img1.png"
    />
    <div
      class="post-content"
    >
      <h2>
        Titulo1
      </h2>
      <p>
        Corpo1
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="Titulo2"
      src="img/img2.png"
    />
    <div
      class="post-content"
    >
      <h2>
        Titulo2
      </h2>
      <p>
        Corpo2
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="Titulo3"
      src="img/img3.png"
    />
    <div
      class="post-content"
    >
      <h2>
        Titulo3
      </h2>
      <p>
        Corpo3
      </p>
    </div>
  </div>
</div>
`;
